ray-curator/ray_curator/stages/deduplication/fuzzy/fuzzy_deduplication.py

```diff
@@ -0,0 +1,288 @@
+        self.use_64_bit_hash = use_64_bit_hash
+        self.bands_per_iteration = bands_per_iteration
+
+        self.env_vars = env_vars
+
+        self.num_hashes = self.num_bands * self.minhashes_per_band
+        self.executor_config = {"runtime_env": {"env_vars": env_vars}} if env_vars is not None else None
+
+        self._validate_inputs()
+
+    def _validate_inputs(self) -> None:
+        if self.char_ngrams < 20:  # noqa: PLR2004
+            logger.warning(
+                "Using a small char_ngrams value might lead to a large number (~5%) of false positives during deduplication."
+                " Using a value of at least 20 for char_ngrams is recommended.",
+            )
+        if self.perform_removal:
+            msg = "Removal is not implemented yet"
+            raise NotImplementedError(msg)
+        if self.bands_per_iteration < 1 or self.bands_per_iteration > self.num_bands:
+            msg = "bands_per_iteration must be between [1, num_bands]"
+            raise ValueError(msg)
+        if self.output_path is None and self.perform_removal:
+            msg = "output_path must be provided if perform_removal is True"
+            raise ValueError(msg)
+        if not self.perform_removal and self.output_path is not None:
+            logger.warning("output_path will be unused as perform_removal is False")
+
+    def _generate_minhash_pipeline(self, generate_input_filegroups: bool) -> Pipeline:
+        stages = []
+        if generate_input_filegroups:
+            stages.append(
+                FilePartitioningStage(
+                    file_paths=self.input_path,
+                    file_extensions=self.input_file_extensions,
+                    blocksize="1GiB",  # compressed size of 1GiB
+                    storage_options=self.read_kwargs.get("storage_options") if self.read_kwargs is not None else None,
+                ),
+            )
+        stages.append(
+            MinHashStage(
+                output_path=self.cache_path,
+                text_field=self.text_field,
+                char_ngrams=self.char_ngrams,
+                num_hashes=self.num_hashes,
+                seed=self.seed,
+                use_64bit_hash=self.use_64_bit_hash,
+                read_format=self.input_filetype,
+                read_kwargs=self.read_kwargs,
+                write_kwargs=self.cache_kwargs,
+            ),
+        )
+        return Pipeline(
+            name="minhash_pipeline",
+            stages=stages,
+        )
+
+    def _generate_lsh_pipeline(self) -> Pipeline:
+        cache_dir_fs = get_fs(self.cache_path, self.cache_kwargs)
+        return Pipeline(
+            name="lsh_duplicate_identification_pipeline",
+            stages=[
+                FilePartitioningStage(
+                    file_paths=cache_dir_fs.sep.join([self.cache_path, "MinHashStage"]),
+                    file_extensions=[".parquet"],
+                    blocksize="2GiB",
```

Comment:
Similar comment here, and I was wondering why they are different?

Reply:
The prev fuzzy dedup impl would read the minhashes with a blocksize of `2GiB` before shuffling. This matches the same pattern. 
Side note: RapidsMPFShuffle is sensitive to the number of partitions so a larger blcocksize is preferred here